processor/src/decoder/mod.rs

```diff
@@ -45,6 +53,45 @@ where
     // JOIN BLOCK
     // --------------------------------------------------------------------------------------------
 
+    /// Starts decoding of a JOIN node.
+    pub(super) fn start_join_node(
+        &mut self,
+        node: &JoinNode,
+        mast_forest: &MastForest,
+    ) -> Result<(), ExecutionError> {
```

Comment:
I'm wondering if we should make this method (and other similar methods) oblivious to the fact that `mast_forest` exists. For example, this method could look like:
```Rust
pub(super) fn start_join_node(
    &mut self,
    node: &JoinNode,
    children: [&MastNode; 2],
) -> Result<(), ExecutionError>
```
And then the job of fetching children of a given node would reside in the `Process` struct.

Reply:
> And then the job of fetching children of a given node would reside in the Process struct.

Note that `start_join_node()` is also a method of `Process`.

I personally prefer how it is now, since it makes `execute_join_node()` (and others) [very clean](https://github.com/0xPolygonMiden/miden-vm/blob/5b90c5b5380a4fd2e90ea914350b503f01a30fa6/processor/src/lib.rs#L267-L273). Needing to fetch the children there would obfuscate that nice high-level view of what executing a JOIN node looks like.